ast/ast.go

```diff
@@ -255,20 +255,25 @@ type (
 	}
 	// A GroupByProc node represents a proc that consumes all the records
 	// in its input, partitions the records into groups based on the values
-	// of the fields specified in the keys paramater, and applies one or
-	// more reducers to each group.  If the duration parameter is non-zero,
-	// then the groups further partioned by time according to the time interval
-	// equal to the duration.  In this case, the proc transmits to its output
-	// the reducer results from each time interval as they complete so that
-	// large time ranges are processed and streamed efficiently.
-	// The limit parameter specifies the number of different groups that can
-	// be aggregated over. When absent, the runtime defaults to an appropriate value.
+	// of the fields specified in the keys field (where the first key is the
+	// primary key), and applies reducers (if any) to each group. If the
+	// duration field is non-zero, then the groups further partioned by time
+	// into bins of the duration. In this case time is the primary key.
+	// The Limit field specifies the number of different groups that can be
+	// aggregated over. When absent, the runtime defaults to an
+	// appropriate value.
+	// The InputSortDir field indicates that input is sorted (with
+	// direction indicated by the sign of the field) in the
+	// primary key. In this case, the proc outputs the reducer
+	// results from each key as they complete so that large inputs
+	// are processed and streamed efficiently.
```

Comment:
```suggestion
	// of the fields specified in the keys field (where the first key is the
	// grouping key), and applies reducers (if any) to each group.  If the
	// Duration field is non-zero, then the groups are further partitioned by time
	// into bins of the duration. In this case, the grouping key is always ts.
	// The Limit field specifies the number of different groups that can be
	// aggregated over. When absent, the runtime defaults to an
	// appropriate value.
	// The InputSortDir field indicates that input is sorted (with
	// direction indicated by the sign of the field) in the
	// grouping key. In this case, the proc outputs the reducer
	// results from each key as they complete so that large inputs
	// are processed and streamed efficiently.
```

Reply:
Applied but defined first key as "primary grouping key" rather than "grouping key" because all keys are grouping keys.